File: essential-100/src/bin/10.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        a: [usize; n],
        q: usize,
        m: [usize; q],
    }

    'outer: for &mi in m.iter() {
        for bit in 0..(1 << n) {
            let bs = format!("{:b}", bit);
            let sums =
                bs.chars().rev().enumerate().fold(
                    0,
                    |acc, (i, e)| {
                        if e == '1' {
                            acc + a[i]
                        } else {
                            acc
                        }
                    },
                );
            if sums == mi {
                println!("Yes");
                continue 'outer;
            }
        }
        println!("No");
    }
}

File: ESSENTIAL_100/10.py
n = int(input())
a = list(map(int, input().split()))
q = int(input())
M = list(map(int, input().split()))


def solve(m):
    for bit in range(1 << n):
        sum = 0
        for i in range(n):
            if bit & 1 << i:
                sum += a[i]
        if sum == m:
            return True
    return False


for m in M:
    print("yes" if solve(m) else "no")
